SlaveWAMPServer.spec.js

```diff
@@ -211,10 +209,305 @@ describe('SlaveWAMPServer', () => {
+				workerId: 0,
 			})).to.be.ok();
 
 			expect(workerMock.sendToMaster.called).not.to.be.ok();
 		});
 	});
+
+	describe('sendToMaster', () => {
+		let validProcedure;
+		let validData;
+		let validSocketId;
+		let validSignature;
+		const actionCb = sinon.spy();
+
+		beforeEach(() => {
+			actionCb.reset();
+			validProcedure = 'validProcedure';
+			validData = { validKey: 'validValue' };
+			validSocketId = 'validSocketId';
+			validSignature = `${new Date().getTime()}_0`;
+		});
+
+		before(() => {
+			sinon.stub(Math, 'random').returns(0);
+		});
+
+		it('should pass correct InterProcessRPCRequestSchema compatible request to sendToMaster function', () => {
+			slaveWampServer.sendToMaster(validProcedure, validData, validSocketId, actionCb);
+			expect(workerMock.sendToMaster.calledOnce).to.be.ok();
+			expect(workerMock.sendToMaster.calledWith({
+				type: '/InterProcessRPCRequestSchema',
+				procedure: validProcedure,
+				data: validData,
+				socketId: validSocketId,
+				workerId: 0,
+				signature: validSignature,
+			})).to.be.ok();
+		});
+
+		it('should pass create a new entry in interProcessRPC map', () => {
+			slaveWampServer.sendToMaster(validProcedure, validData, validSocketId, actionCb);
+			expect(slaveWampServer.interProcessRPC).to.have.property(validSocketId);
+			expect(slaveWampServer.interProcessRPC[validSocketId]).to.have.property(validProcedure);
+			expect(slaveWampServer.interProcessRPC[validSocketId][validProcedure]).to.have.property(validSignature).to.be.a('function');
+		});
+	});
+
+	describe('onSocketDisconnect', () => {
+		let validSocketId;
+		let validProcedure;
+		let validSignature;
+
+		beforeEach(() => {
+			validSocketId = 'validSocketId';
+			validProcedure = 'validProcedure';
+			validSignature = 'validSignature';
+		});
+
+		it('should leave interProcessRPC in initial state when invoked without arguments', () => {
+			slaveWampServer.onSocketDisconnect();
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should leave interProcessRPC in initial state when entry does not exist', () => {
+			slaveWampServer.onSocketDisconnect(validSocketId);
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should remove existing entry from interProcessRPC', () => {
+			slaveWampServer.interProcessRPC = {
+				[validSocketId]: {
+					[validProcedure]: {
+						[validSignature]: () => {},
+					},
+				},
+			};
+			slaveWampServer.onSocketDisconnect(validSocketId);
+			expect(slaveWampServer.interProcessRPC).to.be.empty();
+		});
+	});
+
+	describe('saveCall', () => {
+		let validSocketId;
+		let validProcedure;
+		let validSignature;
+		let validRequest;
+		let validCb;
+
+		beforeEach(() => {
+			validSocketId = 'validSocketId';
+			validProcedure = 'validProcedure';
+			validSignature = 'validSignature';
+			validRequest = {
+				socketId: validSocketId,
+				procedure: validProcedure,
+				signature: validSignature,
+			};
+			validCb = () => {};
```

Comment:
This can probably go in a `const` since it's not a spy, assuming we won't be mutating callback functions.